persist/sqlite/volumes.go

```diff
@@ -526,12 +526,19 @@ func emptyLocation(tx *txn) (loc storage.SectorLocation, err error) {
 // emptyLocationForMigration returns an empty location in a writable volume. If there is no
 // space available, ErrNotEnoughStorage is returned.
 func emptyLocationForMigration(tx *txn, volumeID int64, maxIndex uint64) (loc storage.SectorLocation, err error) {
+	loc, err = emptyLocation(tx)
+	if !errors.Is(err, storage.ErrNotEnoughStorage) {
+		return // either a db error or a valid location
+	} else if maxIndex == 0 {
+		err = storage.ErrNotEnoughStorage
+		return // no space in current volume
+	}
```

Comment:
Did you mean to exit early when `emptyLocation` works? As the code is right now I'm slightly confused because both `emptyLocation` as well as `emptyLocationForMigration` increment `sector_writes` and you don't ever use the result of `emptyLocation` but still don't do `_, err = emptyLocation(tx)` to indicate all you are actually interested in is the `err`.